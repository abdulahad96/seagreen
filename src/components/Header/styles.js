import {StyleSheet} from 'react-native';
import {Metrics, Colors, Fonts} from '../../theme';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.Ebony_Clay,
    height: Metrics.screenHeight * 0.095,
    flexDirection: 'row'
  },
  containerShadow: {
    shadowColor: Colors.Black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  menuImage: {
    width: Metrics.ratio(25),
    height: Metrics.ratio(25),
  },
  TouchableMenu: {
    width: Metrics.ratio(50),
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerText: {
    width: Metrics.screenWidth - Metrics.doubleBaseMargin,
    textAlign: 'center',
    alignSelf: 'center',
    fontSize: Metrics.ratio(18),
    fontFamily: Fonts.type.RobotoBold,
    color: Colors.white,
  },
});
