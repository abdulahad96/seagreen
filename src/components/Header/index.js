import React, { Component } from 'react';
import { Text, View, TouchableOpacity, Image } from 'react-native';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './styles';

import { Metrics, Colors } from '../../theme';

const Header = ({
  headerText,
  headerTextStyle,
  headerShadow,
  leftIcon,
  leftIconStyle,
  leftBtnPress,
  rightIcon,
  rightIconStyle,
  rightBtnPress,
  rightIconImageStyle,
  leftIconImageStyle,
  notification,
  headerCustom

}) => {
 

  return (
    <View
    style={[
      styles.container,
      headerShadow && styles.containerShadow,
      headerCustom,
     
    ]}>
   
    <TouchableOpacity
      onPress={leftBtnPress}
      style={[styles.TouchableMenu
      , leftIconStyle
      ]}
      testID={'header-left-btn'}
      >
        <Image style={leftIconImageStyle} source={leftIcon}/>
       </TouchableOpacity>

    <Text style={[styles.headerText, headerTextStyle]}>{headerText}</Text>

    <TouchableOpacity
      onPress={rightBtnPress}
      style={[styles.TouchableMenu,
       rightIconStyle
       ]}
       testID={'header-right-btn'}
       >
         <Image 
         style={[styles.rightIconImage,rightIconImageStyle]} 
         source={rightIcon}/>
         {notification > 0 && <View style={{position:"absolute",top:Metrics.ratio(5),right:Metrics.ratio(10),backgroundColor:"#6f6a41",paddingVertical:Metrics.ratio(2),paddingHorizontal:Metrics.ratio(5),borderRadius:Metrics.ratio(50)}}>
           <Text style={{fontSize:Metrics.ratio(8),color:'white'}}>{notification}</Text>
           </View>}
      
    </TouchableOpacity>
  </View>
  );
}

Header.propTypes = {
    headerText: PropTypes.string,
    headerTextStyle: PropTypes.object,
    headerShadow: PropTypes.bool,
    leftIcon: PropTypes.string,
    leftIconStyle: PropTypes.object,
    leftBtnPress: PropTypes.func,
    rightIcon: PropTypes.string,
    rightIconStyle: PropTypes.object,
    rightBtnPress: PropTypes.func,
    rightIconImageStyle:PropTypes.object,
    leftIconImageStyle:PropTypes.object,
    notification:PropTypes.number

};

Header.defaultProps = {
    headerText: '',
    headerTextStyle: undefined,
    headerShadow: true,
    leftIconStyle: undefined,
    leftIconSize: undefined,
    leftBtnPress: undefined,
    rightIconStyle: undefined,
    rightIconSize: undefined,
    rightBtnPress: undefined,
    rightIconImageStyle:undefined,
    leftIconImageStyle:undefined,
    notification:undefined
};
export default Header;
