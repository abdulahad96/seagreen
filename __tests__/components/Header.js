import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import Header from '../../src/components/Header';

describe('Component Header', () => {
    const leftBtnPress = jest.fn()
    const rightBtnPress = jest.fn()
    test('renders correctly', () => {
        const output = render(<Header
            headerText={''}
        />);
        expect(output).toMatchSnapshot();
    });
    test('renders correctly with leftIcon and rightIcon and Header Text', () => {
        const output = render(<Header
            headerText={''}
            leftIcon={''}
            rightIcon={''}
        />);
        expect(output).toMatchSnapshot();
    });
    test('renders correctly with notification', () => {
        const output = render(<Header
            headerText={''}
            leftIcon={''}
            rightIcon={''}
            notification={1}
        />);
        expect(output).toMatchSnapshot();
    });

    test('renders correctly with notification', () => {
        const output = render(<Header
            headerText={''}
            leftIcon={''}
            rightIcon={''}
            notification={1}
        />);
        expect(output).toMatchSnapshot();
    });
    test('renders when left icon is pressed', () => {
        const { getByTestId } = render(

            <Header
                headerText={''}
                leftIcon={''}
                rightIcon={''}
                notification={1}
                leftBtnPress={leftBtnPress}

            />
        );
        fireEvent(getByTestId('header-left-btn'), 'press');
        expect(leftBtnPress).toHaveBeenCalled();
    });
    test('renders when right icon is pressed', () => {
        const { getByTestId } = render(

            <Header
                headerText={''}
                leftIcon={''}
                rightIcon={''}
                notification={1}
                rightBtnPress={rightBtnPress}

            />
        );
        fireEvent(getByTestId('header-right-btn'), 'press');
        expect(rightBtnPress).toHaveBeenCalled();
    });
});

