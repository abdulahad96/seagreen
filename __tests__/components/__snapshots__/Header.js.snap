// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flexDirection": "row",
        "height": 126.73,
      },
      Object {
        "elevation": 5,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-left-btn"
  >
    <Image />
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontFamily": "Roboto-Bold",
          "fontSize": 36,
          "textAlign": "center",
          "width": 550,
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-right-btn"
  >
    <Image
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Component Header renders correctly with leftIcon and rightIcon and Header Text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flexDirection": "row",
        "height": 126.73,
      },
      Object {
        "elevation": 5,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-left-btn"
  >
    <Image
      source=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontFamily": "Roboto-Bold",
          "fontSize": 36,
          "textAlign": "center",
          "width": 550,
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-right-btn"
  >
    <Image
      source=""
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Component Header renders correctly with notification 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flexDirection": "row",
        "height": 126.73,
      },
      Object {
        "elevation": 5,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-left-btn"
  >
    <Image
      source=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontFamily": "Roboto-Bold",
          "fontSize": 36,
          "textAlign": "center",
          "width": 550,
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-right-btn"
  >
    <Image
      source=""
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#6f6a41",
          "borderRadius": 100,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
          "position": "absolute",
          "right": 20,
          "top": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component Header renders correctly with notification 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flexDirection": "row",
        "height": 126.73,
      },
      Object {
        "elevation": 5,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-left-btn"
  >
    <Image
      source=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontFamily": "Roboto-Bold",
          "fontSize": 36,
          "textAlign": "center",
          "width": 550,
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="header-right-btn"
  >
    <Image
      source=""
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#6f6a41",
          "borderRadius": 100,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
          "position": "absolute",
          "right": 20,
          "top": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;
